overlord/snapstate/snapstate.go

```diff
@@ -285,3 +285,27 @@ func Set(s *state.State, name string, snapst *SnapState) {
 	snaps[name] = &raw
 	s.Set("snaps", snaps)
 }
+
+// All returns information about all active snaps.
+func All(s *state.State) ([]*snap.Info, error) {
+	var stateMap map[string]*SnapState
+	var infos []*snap.Info
+	if err := s.Get("snaps", &stateMap); err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	for snapName, snapState := range stateMap {
+		if !snapState.Active {
+			continue
+		}
+		if len(snapState.Sequence) == 0 {
+			continue // XXX: is this possible?
```

Comment:
Actually, sorry, I take that back.. no need to check. It's active, so there must be something in Sequence or there's something seriously broken.